src/DETHRACE/common/sound.c

```diff
@@ -681,19 +681,59 @@ int GetIndexFromOutlet(tS3_outlet_ptr pOutlet) {
 // IDA: int __usercall DRS3StartCDA@<EAX>(tS3_sound_id pCDA_id@<EAX>)
 int DRS3StartCDA(tS3_sound_id pCDA_id) {
     LOG_TRACE("(%d)", pCDA_id);
-    NOT_IMPLEMENTED();
+
+    if (!gCD_is_disabled && gMusic_available) {
+        if (!gCDA_is_playing && !gCDA_tag) {
+            if (S3CDAEnabled()) {
+                S3StopOutletSound(gMusic_outlet);
+                if (gSound_enabled) {
+                    if (gProgram_state.cockpit_on && gProgram_state.cockpit_image_index >= 0) {
+                        S3Service(1, 0);
+                    } else {
+                        S3Service(0, 0);
+                    }
+                    if (pCDA_id == 9999) {
+                        do {
+                            pCDA_id = gRandom_CDA_tunes[IRandomBetween(0, 7)];
```

Comment:
dethrace's `IRandomBetween(0, 7)` returns variable in the range `[0, 7]` (inclusive 0 and 7).
However, `gRandom_CDA_tunes` has length 7, with maximum index 6.

The `MUSIC` folder of my gog copy has 8 tracks (+ a 4s silence track) so I think changing `gRandom_CDA_tunes` to `int[8]` makes sense.

Reply:
The 7th item on my system is currently 0, which causes a stack overflow in `S3GetDescriptorByID(0)` -> `S3GetDescriptorByID(0)` -> `S3GetDescriptorByID(0)` -> ...

Reply:
Appreciate your second pair of eyes on this one! I originally made the change to `[8]`, based on there being 8 tracks in `SOUND.TXT`, and seeing `9607` in IDA at the end of the array. When I saw the last track was silence, I figured it was deliberately left off. 

I didn't think about the `IRandomBetween(0, 7)` though! :) 

I wonder if the original code was declared as 7 but initialized with 8 items like ```int gRandom_CDA_tunes[7] = { 9600, 9601, 9602, 9603, 9604, 9605, 9606, 9607 };``` maybe